File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int mx=0;
        int cnt=0;
        for(auto x:s){
            if(x=='('){
                cnt++;
                mx=max(mx,cnt);
            }else if(x==')'){
                cnt--;
            }else{
                continue;
            }
        }
        return mx;
    }
};

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int n=s.length();
        int ans=0;
        int dep=0;

        for(int i=0;i<n;i++){
            if(s.charAt(i)=='('){
                dep++;
                ans=Math.max(ans,dep);
            }else if(s.charAt(i)==')'){
                dep--;
            }
        }
        return ans;
    }
}